src/driver/op.rs

```diff
@@ -63,7 +62,7 @@ impl<T> Op<T> {
     /// the kernel.
     pub(super) fn submit_with<F>(data: T, f: F) -> io::Result<Op<T>>
     where
-        F: FnOnce() -> squeue::Entry,
+        F: FnOnce(&mut T) -> squeue::Entry,
```

Comment:
Why mut?

Reply:
Because submitting certain ops might mutate the stored data.